server/src/main/java/io/crate/exceptions/ColumnUnknownException.java

```diff
@@ -27,9 +27,10 @@
 import java.util.Collections;
 import java.util.Locale;
 
+import org.elasticsearch.ElasticsearchException;
 import org.jetbrains.annotations.NotNull;
 
-public class ColumnUnknownException extends RuntimeException implements ResourceUnknownException, TableScopeException {
+public class ColumnUnknownException extends ElasticsearchException implements ResourceUnknownException, TableScopeException {
```

Comment:
What's the purpose of this? I'd say please revert.

Reply:
Without this, the new test would throw `NotSerializableExceptionWrapper` instead:

```
    @Test
     public void test_aliased_unknown_object_key() {
         try (var session = sqlExecutor.newSession()) {
             execute("create table test (o object)", session);
             execute("insert into test values({a=1})", session);
             refresh();

             session.sessionSettings().setErrorOnUnknownObjectKey(true);
             assertThatThrownBy(() -> execute("select T.o['unknown'] from (select * from test) T", session))
                 .isExactlyInstanceOf(ColumnUnknownException.class)
                     .hasMessage("The object `{a=1}` does not contain the key `unknown`");

             session.sessionSettings().setErrorOnUnknownObjectKey(false);
             execute("select T.o['unknown'] from (select * from test) T", session);
             assertThat(response).hasRows("NULL");
         }
     }
```

Reply:
Then please use `Asserts.assertSQLError(() -> execute(....` instead.
I don't think it's good idea to change that now to EsException.

Reply:
Thanks for suggestion, changing to `Asserts.assertSQLError` did not prevent `NotSerializableExceptionWrapper`. I will think about this further.

Reply:
The error code still flip-flops between 4043 and 5000 (due to `NotSerializableExceptionWrapper`)

```
Asserts.assertSQLError(() -> execute("select T.o['unknown'] from (select * from test) T", session))
                .hasPGError(UNDEFINED_COLUMN)
                .hasHTTPError(NOT_FOUND, 4043)
                .hasMessage("The object `{a=1}` does not contain the key `unknown`");
```

Reply:
What do you think about this @mfussenegger ? If we keep the[ error behaviour](https://github.com/crate/crate/pull/14371#discussion_r1262645775), I think we have no other choice but to serialize `ColumnUnknownException`. (did not add the complete logic to be fully serializable yet)

Reply:
Why isn't the error already triggering on the coordinator/handler?

Reply:
Instead of throwing an exception we are continuing the execution as a subscript function, so we can no longer identify that the object key is unknown until executing the function, https://github.com/crate/crate/pull/14371/files#diff-a17636694f46dde4f5ad1e95417d763f36cd0c55b9e28b700800dadb815fc0d5R2996-R2999

Reply:
Raising an error earlier if `errorOnUnknownObjectKey` is set isn't possible?

In any case, making the exception serializable seems okay to me.

Reply:
Sorry it was my confusion I will try to go without serializing it.